### PR TITLE
EZP-28950: Show how to change table character set

### PR DESCRIPTION
**Pull request by glye**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-28950](https://jira.ez.no/browse/EZP-28950)
| **Bug/Improvement**| no
| **New feature**    | no
| **Target version** | master / 7.2
| **BC breaks**      | kind of
| **Tests pass**     | yes
| **Doc needed**     | yes

https://github.com/ezsystems/ezpublish-kernel/commit/0f82a088ebe7949970f9cf1acb687b21b42a0779 did not contain statements or info on how to change table charset on existing tables, when upgrading to 7.2.

Here I explain how to do it, rather than provide the complete set of SQL commands, because I want administrators to think carefully about which tables to convert, and which collation to use, rather than just copy a bunch of SQL commands. Also, because the number of tables varies depending on whether you use eZ Platform or Enterprise Edition, and on whether you have any additional custom tables.

### CODE REVIEW

File: doc/upgrade/7.2.md [hunk 1]
```diff
@@ -4,9 +4,18 @@ See `doc/bc/changes-7.2.md` for requirements changes and deprecations.
...
 
-Beware that these upgrade statements may fail due to index collisions. This is because the indexes have been shortened, so duplicates may occur. If that happens, you must remove the duplicates manually, and then repeat the statements that failed.
+After successfully running those statements, change the character set of each table to `utf8mb4`. This can be done as shown below, for each of the standard tables in eZ Platform, and for any custom table you may want to convert. You may specify a collation; if you don't, the default collation for utf8mb4 will be used.
+
+To get a list of tables, enter the SQL command `show tables;`. For each of the tables you want to convert, replace `tbl_name` with the name of the table to convert:
+```sql
+-- Example using default collation
+ALTER TABLE tbl_name CONVERT TO CHARACTER SET utf8mb4;
```
Comment (alongosz) [inline doc/upgrade/7.2.md]: @glye Do we need to show an example setting default collation when we really want users to specify `utf8mb4_unicode_520_ci`?

Comment (glye) [reply doc/upgrade/7.2.md]: @alongosz Yeah, I can drop that. I should add a bit more info about collation, too.

**Comment by glye** [comment]
Update: Clearer collation recommendation.

File: doc/upgrade/7.2.md [hunk 2]
```diff
@@ -4,9 +4,15 @@ See `doc/bc/changes-7.2.md` for requirements changes and deprecations.
...
+The character set for MySQL/MariaDB database tables should be changed from `utf8` to `utf8mb4` to support 4-byte characters. See `data/update/mysql/dbupdate-7.1.0-to-7.2.0.sql` for the statements that shorten table indexes, these must be run first. Beware that these upgrade statements may fail due to index collisions, which may occur when indexes are shortened. If that happens, you must remove the duplicates manually, and then repeat the statements that failed.
 
-Beware that these upgrade statements may fail due to index collisions. This is because the indexes have been shortened, so duplicates may occur. If that happens, you must remove the duplicates manually, and then repeat the statements that failed.
+After successfully running those statements, change the character set of each table to `utf8mb4`. This can be done as shown below, for each of the standard tables in eZ Platform, and for any custom table you may want to convert. You may specify a collation; if you don't, the default collation for utf8mb4 will be used. We recommend specifying `utf8mb4_unicode_520_ci`, which supports emoji collation (the collation to use depends on your needs).
```
Comment (alongosz) [inline doc/upgrade/7.2.md]: Hmm, now I'm gonna nitpick on the last sentence here :grinning: 
Wasn't this supposed to be 
> We recommend specifying `utf8mb4_unicode_520_ci` collation, which supports e.g. emojis

Comment (glye) [reply doc/upgrade/7.2.md]: @alongosz `utf8mb4_unicode_520_ci` is probably not right for everyone, hence the parentheses remark. Some may want binary, for example.

Comment (alongosz) [reply doc/upgrade/7.2.md]: @glye sorry, I meant the part that stands before parentheses. Just updated comment to make it more clear.
I don't get "emoji collation" :)

Comment (glye) [reply doc/upgrade/7.2.md]: @alongosz Right. I was trying to give a hint while not being too wordy. Any collation "supports" emojis (allows them to be used). By "emoji collation" I mean "is able to tell the difference between different emojis". The standard collation thinks all emojis are the same.

How about:
> We recommend specifying `utf8mb4_unicode_520_ci`, which is able to tell different emojis apart, though you may have other requirements.

Comment (alongosz) [reply doc/upgrade/7.2.md]: ^ Ok, that sounds good.

**Comment by glye** [comment]
Update: Clearer-er collation recommendation.
